server/src/main/java/org/apache/cassandra/sidecar/utils/AuthUtils.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.utils;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import io.vertx.ext.auth.User;
+import org.apache.cassandra.sidecar.acl.authorization.Permission;
+import org.apache.cassandra.sidecar.acl.authorization.StandardPermission;
+import org.apache.cassandra.sidecar.acl.authorization.WildcardPermission;
+
+import static org.apache.cassandra.sidecar.acl.authorization.WildcardPermission.WILDCARD_PART_DIVIDER_TOKEN;
+import static org.apache.cassandra.sidecar.acl.authorization.WildcardPermission.WILDCARD_TOKEN;
+
+/**
+ * Class with utility methods for Authentication and Authorization.
+ */
+public class AuthUtils
+{
+    /**
+     * Extracts a list of identities a user holds from their principal.
+     *
+     * @param user User object in Vertx
+     * @return extracted identities of user
+     */
+    public static List<String> extractIdentities(User user)
+    {
+        if (user.principal() == null)
+        {
+            return Collections.emptyList();
+        }
+
+        if (!user.principal().containsKey("identity") && !user.principal().containsKey("identities"))
+        {
+            return Collections.emptyList();
+        }
+
+        return Optional.ofNullable(user.principal().getString("identity"))
+                       .map(Collections::singletonList)
+                       .orElseGet(() -> Arrays.asList(user.principal()
+                                                          .getString("identities")
+                                                          .split(",")));
+    }
```

Comment:
What if principal has both `identity` and `identities`? This implementation returns early when `identity` exists. 

```suggestion
    public static List<String> extractIdentities(User user)
    {
        JsonObject principal = user.principal();

        if (principal == null)
        {
            return Collections.emptyList();
        }

        List<String> identities = new ArrayList<>();
        if (principal.containsKey("identity"))
        {
            identities.add(principal.getString("identity"));
        }

        if (principal.containsKey("identities"))
        {
            String[] parts = user.principal().getString("identities").split(",");
            identities.addAll(Arrays.asList(parts));
        }

        return identities;
    }
```